subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/DirectoryFileTreeTest.groovy

```diff
@@ -234,23 +263,26 @@ class DirectoryFileTreeTest extends AbstractProjectBuilderSpec {
 
         private void handleDetails(FileVisitDetails details) {
             def file = details.getFile()
-            assert expectedVisitations[0].contains(file)
-            expectedVisitations[0].remove(file)
-            if (expectedVisitations[0].isEmpty() || file == stopOn) {
-                expectedVisitations.remove(0)
-                if (file == stopOn) {
-                    details.stopVisiting()
-                }
+            def found = expectedVisitations.find { it.remove(file) }
+            if (file == stopOn) {
+                expectedVisitations.remove(found)
+                details.stopVisiting()
+            } else if (found.isEmpty()) {
+                expectedVisitations.remove(found)
```

Comment:
❌ Does it make sense any more to provide a list of lists of expected visitations here if we don't expect the order anyway? We could directly use a collecting file visitor and then assert that visited files at the end.

Reply:
I didn't make sense in the beginning either, just didn't want to refactor it too much. Done